bot/exts/info/information.py

```diff
@@ -537,17 +549,44 @@ async def rules(self, ctx: Context, rules: Greedy[int]) -> None:
 
         # Remove duplicates and sort the rule indices
         rules = sorted(set(rules))
+        # Remove duplicate keywords
+        keywords = set(keywords)
 
         invalid = ", ".join(str(index) for index in rules if index < 1 or index > len(full_rules))
 
-        if invalid:
+        if invalid and not keywords:
             await ctx.send(shorten(":x: Invalid rule indices: " + invalid, 75, placeholder=" ..."))
             return
 
-        for rule in rules:
-            self.bot.stats.incr(f"rule_uses.{rule}")
-
-        final_rules = tuple(f"**{pick}.** {full_rules[pick - 1]}" for pick in rules)
+        final_rules = set()
+
+        for pick in rules:
+            self.bot.stats.incr(f"rule_uses.{pick}")
+            final_rules.add(
+                f"**{pick}.** {full_rules[pick - 1][0]}\n\n"
+                f"You can also invoke this rule with the following keywords: "
+                f"{', '.join(full_rules[pick -1][1])}")
+
+        for keyword in keywords:
+            for pick, rule in enumerate(full_rules):
+                if keyword not in rule[1]:
+                    continue
```

Comment:
As I stated in a previous comment, I do have concerns about supporting multiple keywords and also allowing users to write sentences after the command. To avoid unintentional interpretation of words as keywords, I think this argument parser should stop when it encounters an invalid keyword. I believe this will require the keywords to not be converted into a set.

Reply:
Allowing keywords anywhere is an interesting idea, as it would allow things like `!rule that breaks the tos of xyz`. That said, I think I agree and prefer the more conservative way of stopping after an invalid keyword, I could possibly be convinced otherwise though.